WordPress/gutenberg
# Add Typography: text orientation (writing mode)

## What?

Adds a new text orientation option where text can be displayed horizontally or vertically.
Partial for https://github.com/WordPress/gutenberg/issues/50319

## Why?

To allow more design options.

## How?

- A new orientation setting is added to the Typography panel with the help of a writing-mode-control component.
- It has two options: Horizontal and vertical, that each adds inline CSS for `writing-mode` through the Style Engine.
- A writing mode block support is added under typography, and it is disabled by default.
- The block support is enabled for paragraphs and post navigation links in block.json.

## Testing Instructions
This setting is disabled by default and can be enabled for themes that include theme.json.

1. In theme.json, under settings > typography, add `writingMode: true`
2. Under styles > blocks, add
```
"core/paragraph": {
	"typography": {
		"writingMode": "vertical-rl"
	}
}
```
3. Activate the theme. Confirm that all paragraphs are vertical.
4. Remove the block style from theme.json.
5. Go to the Site Editor, Styles, Blocks, Paragraph, Typography, and set the orientation to vertical. Confirm that the block is vertical.
6. Change the orientation to horizontal and confirm the change is applied.
7. Next, select a paragraph block and find the orientation setting in the block settings sidebar, typography panel. It should not be displayed by default, but be available from the three dot menu.
8. Confirm that both options work in the editor and front.
9. Change the site language to use an RTL language, and repeat step 7. Confirm that both options work and that the text is facing the correct direction.

### Testing Instructions for Keyboard


## Screenshots or screencast 
https://github.com/WordPress/gutenberg/assets/7422055/1d222d1a-99d7-4db9-b803-4649c104b6a4
